cloud-sql/sqlserver/tedious/README.md

```diff
@@ -0,0 +1,387 @@
+[Node.js on Cloud Run samples](../../../run/).
+
+### Deploy to Cloud Functions
+
+To deploy the service to [Cloud Functions](https://cloud.google.com/functions/docs) run the following command:
+
+```sh
+gcloud functions deploy votes --gen2 --runtime nodejs18 --trigger-http \
+  --allow-unauthenticated \
+  --entry-point votes \
+  --region <INSTANCE_REGION> \
+  --set-env-vars INSTANCE_CONNECTION_NAME=<PROJECT_ID>:<INSTANCE_REGION>:<INSTANCE_NAME> \
+  --set-env-vars DB_USER=$DB_USER \
+  --set-env-vars DB_PASS=$DB_PASS \
+  --set-env-vars DB_NAME=$DB_NAME
+```
+
+## Cloud SQL Auth Proxy Usage
+
+### Running locally
+
+You may optionally download and install the `cloud-sql-proxy` by
+[following the
+instructions](https://cloud.google.com/sql/docs/sqlserver/sql-proxy#install) as an
+alternative to using the
+[Cloud SQL Node.js Connector](https://github.com/GoogleCloudPlatform/cloud-sql-nodejs-connector).
+
+### Launch proxy with TCP
+
+To run the sample locally with a TCP connection, set environment variables and
+launch the proxy as shown below.
+
+#### Linux / Mac OS
+
+Use these terminal commands to initialize environment variables:
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS='/path/to/service/account/key.json'
+export INSTANCE_HOST='127.0.0.1'
+export DB_PORT='1433'
+export DB_USER='<DB_USER_NAME>'
+export DB_PASS='<DB_PASSWORD>'
+export DB_NAME='<DB_NAME>'
+```
+
+Then use this command to launch the proxy in the background:
+
+```bash
+./cloud-sql-proxy <PROJECT_ID>:<INSTANCE_REGION>:<INSTANCE_NAME> &
+```
+
+#### Windows/PowerShell
+
+Use these PowerShell commands to initialize environment variables:
+
+```powershell
+$env:GOOGLE_APPLICATION_CREDENTIALS="<CREDENTIALS_JSON_FILE>"
+$env:INSTANCE_HOST="127.0.0.1"
+$env:DB_PORT="1433"
+$env:DB_USER="<DB_USER_NAME>"
+$env:DB_PASS="<DB_PASSWORD>"
+$env:DB_NAME="<DB_NAME>"
+```
+
+Then use this command to launch the proxy in a separate PowerShell session:
+
+```powershell
+Start-Process -filepath "C:\<path to proxy exe>" -ArgumentList "-instances=<project-id>:<region>:<instance-name>=tcp:1433 -credential_file=<CREDENTIALS_JSON_FILE>"
+```
+
+### Testing the application
+
+1. Next, install the Node.js packages necessary to run the app locally by
+   running the following command:
+
+    ```sh
+    npm install
+    ```
+
+2. Run the sample app locally with the following command:
+
+    ```sh
+    npm start
+    ```
+
+Navigate towards `http://127.0.0.1:8080` to verify your application is running
+correctly.
+
+### Deploy to Google App Engine Standard
+
+To run on GAE-Standard, create an App Engine project by following the setup for
+these
+[instructions](https://cloud.google.com/appengine/docs/standard/nodejs/quickstart#before-you-begin).
+
+First, update [`app.standard.yaml`](app.standard.yaml) with the correct values
+to pass the environment variables into the runtime. e.g:
+
+```
+env_variables:
+  INSTANCE_HOST: 172.17.0.1
```

Comment:
that IP address is only for GAE flexible I believe, normal localhost IP should work for GAE standard
```suggestion
  INSTANCE_HOST: 127.0.0.1
```